content/posts/0.23-release.md

```diff
@@ -0,0 +1,153 @@
+enhancements](https://docs.python.org/3/whatsnew/3.11.html) is also the first
+version to officially support the WebAssembly as a [PEP11 Tier
+3](https://peps.python.org/pep-0011/#tier-3) platform.
+
+For year Pyodide has been distributing a custom build of CPython with many patches.
+Now that we can rely on a mostly unmodified CPython, it has become significantly easier
+to update CPython versions which is very positive for the
+project's sustainability. We are very grateful for the CPython maintainers
+[¹](#Acknowledgments) who made this possible.
+
+Python 3.11 is estimated to be between 10-60% faster than Python 3.10 for
+native execution, and we see comparable improvements with the WebAssembly
+runtime on our benchmark suite.
+
+## Load time and size optimizations
+
+Reducing the load time and size of a minimal web page using Pyodide has been a
+continuous effort. Several techniques have been implemented aiming to reduce
+download size,
+ - tests and some standard library modules are unvendored when Pyodide is
+   initialized with `loadPyodide({..., fullStdLib=False})`, the default
+   currently, and need to be explicitly loaded. In this release we additionally
+   unvendored `_pydecimal` and `pydoc_data` modules.
+ - Pyodide no longer uses Emscripten preload plugin, and `pyodide.asm.data`
+   file is removed, in favor of `python_stdlib.zip`. This would make it easier
+   to adjust the contents of this file if needed, particularly that
+   `loadPyodide` can now load it from a custom location using the `stdLibURL`
+   parameter.
+ - Python standard libraries are now vendored in a zipfile:
+   /lib/python{version}.zip in the in-browser MEMFS file system. If you need to
+   access the standard library source code, you need to unpack the zip file.
+
+To some extent, however, these optimizations have been offset by the
+growth of the Python standard library, and the enabling additional
+functionality over time. 
+
+As can be seen in the following graph, there are size increases for each major
+Python version, with an overall, slowly decreasing trend for the download size,
+
+<img src="download_size_evolution.webp" alt="Download size evolution for the Python REPL" />
+
+In this release we have also added a py-compiled build for the Python standard
+library and packages. This leads to faster initialization times, however
+because the size of .pyc files increased in Python 3.11
+([cpython#99554](https://github.com/python/cpython/issues/99554)) the overall
+download size is larger.
+
+--------------------------------------------
+|        | full | py-compiled |
+|--------|------|-------------|
+| Download size of Python + sdlib (Brotli compressed) |  5.3 MB |      6.9 MB    | 
+| Load + initialization time (Firefox) |  1.3 s |      0.87 s    |
+| Load + initialization time (Chrome)  |  0.85 s |      0.46 s    |
+
+Similarly load time improvements x1.2 - x2.0 were observed for larger packages
```

Comment:
That was with Python 3.10, it's really unfortunate that Python 3.11 produces so much larger .pyc files. Hopefully, they will fix some of it in 3.12. I need to re-run those benchmarks.